AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.java

```
@@ -217,4 +225,53 @@ public FunctionItem createFromParcel(Parcel in) {
             void consume(AnkiActivity activity);
         }
     }
+
+    private static class ExceptionReportItem extends Item implements Parcelable{
+
+        private static Long lastClickStamp;
+        final int minIntervalMS = 60000;
+        final String exceptionMessage = "Exception report sent by user manually";
+        String reportMode = getSharedPrefs(getInstance().getApplicationContext()).getString(AnkiDroidApp.FEEDBACK_REPORT_KEY, "");
+
+        public ExceptionReportItem(@StringRes int titleRes, @DrawableRes int iconRes) { super(titleRes, iconRes); }
+
+        @Override
+        public void execute(AnkiActivity activity) {
+            long currentTimestamp = SystemClock.uptimeMillis();
+
```

If we're running under a test client, we don't want this

```java
        if (AdaptionUtil.isUserATestClient()) {
            UIUtils.showThemedToast(activity, activity.getString(R.string.user_is_a_robot), false);
            return;
        }
```

---

Sorry sir, I didn't get this part :sweat_smile: could you please explain it once?

---

Add the code at the start of a method - if the user is running under an automated device such as the Firebase Test Lab, which is randomly pressing buttons, then we don't want to report the exception